backends/ebpf/tests/ptf/test.py

```diff
@@ -587,3 +587,57 @@ def runTest(self):
         pkt[IP].dst = 0x11993355  # mask is 0xFF00FFFF
         testutils.send_packet(self, PORT0, pkt)
         testutils.verify_packet(self, pkt, PORT1)
+
+
+class LPMTableCachePSATest(P4EbpfTest):
+    p4_file_path = "p4testdata/table-cache-lpm.p4"
+    p4c_additional_args = "--table-caching"
+
+    def runTest(self):
+        # TODO: make this additional entry working
+        # self.table_add(table="ingress_tbl_lpm", key=["00:11:22:33:44:50/44"], action=1, data=["11:22:33:44:55:67"])
+        self.table_add(table="ingress_tbl_lpm", key=["00:11:22:33:44:55/48"], action=1, data=["11:22:33:44:55:66"])
+        pkt = testutils.simple_ip_packet(eth_dst="00:11:22:33:44:50")
+        exp_pkt = testutils.simple_ip_packet(eth_dst="11:22:33:44:55:66")
+        exp_pkt[Ether].type = 0x8601
+
+        testutils.send_packet(self, PORT0, pkt)
+        testutils.verify_packet(self, exp_pkt, PORT1)
+
+        # modify cache directly - verify that it is used (modify to NoAction)
+        self.table_update(table="ingress_tbl_lpm_cache",
```

Comment:
I don't get why we update here? Can't we just do get and check if the cached entry has been installed in response to packet received earlier?

Reply:
First, the existence of cache entry does not mean it is used for table lookup. By cache modification it is possible to detect if it was used instead of an original table entry.
Second, `table_update` will fail if cache entry does not exist (I verified this by modifying padding value).
Third, for some reasons `nikss-ctl` fails to parse cache silently (should fall back to `byte by byte` mode but do not do this):
```
value: unknown type direct object instance, ignored       <= this is the original table value which contains action ID
failed to parse entry: No such file or directory          <= probably missing action ID
can't get default entry: table not opened or not exists   <= no map for default entry
{
    "ingress_tbl_lpm_cache": {
        "entries": [],
        "DirectCounter": {}
    }
}'
```

Reply:
Ok, got it. A more clear comment would be nice - it's not clear why we do table update here.